subprojects/instant-execution/src/main/kotlin/org/gradle/instantexecution/fingerprint/InstantExecutionCacheFingerprintWriter.kt

```diff
@@ -127,8 +127,10 @@ class InstantExecutionCacheFingerprintWriter(
 
     private
     fun onChangingValue(changingValue: InstantExecutionCacheFingerprint.ChangingDependencyResolutionValue) {
-        if (closestChangingValue == null || closestChangingValue!!.expireAt > changingValue.expireAt) {
-            closestChangingValue = changingValue
+        synchronized(this) {
```

Comment:
We should probably fix the other event methods that can be called concurrently, which I think is all of them.

Reply:
Thanks, Adam. Will do.